### PR TITLE
Remove passwordEncoder with no db

### LINKED ISSUE #11379: No database options: compilation failure
##### **Overview of the issue**

When choosing no database options with JWT or Session, the compilation failed.
See https://dev.azure.com/hipster-labs/jhipster-daily-builds/_build?definitionId=23&_a=summary

cc @codecholeric

##### **Related issues**

Following https://github.com/jhipster/generator-jhipster/issues/10682 and https://github.com/jhipster/generator-jhipster/pull/11050

##### **JHipster Version(s)**

```
jhipster@0.0.1-SNAPSHOT /home/pgrimaud/tmp/24-nodb
└── generator-jhipster@6.7.1 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "io.github.pascalgrimaud.burger",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.7.1",
    "applicationType": "monolith",
    "baseName": "jhipster",
    "packageName": "io.github.pascalgrimaud.burger",
    "packageFolder": "io/github/pascalgrimaud/burger",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "no",
    "devDatabaseType": "no",
    "prodDatabaseType": "no",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "creationTimestamp": 1582490085085,
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en"],
    "blueprints": []
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment (build 11.0.6+10-post-Ubuntu-1ubuntu118.04.1)
OpenJDK 64-Bit Server VM (build 11.0.6+10-post-Ubuntu-1ubuntu118.04.1, mixed mode, sharing)

git version 2.25.1

node: v12.16.0

npm: 6.13.4

yeoman: 3.1.1

yarn: 1.21.1

Docker version 18.03.0-ce, build 0520e24

docker-compose version 1.21.2, build a133471

INFO! Congratulations, JHipster execution is complete!

### PR DESCRIPTION
**Pull request by Falydoor**
-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

Fix #11379

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
@Falydoor : thanks for fixing this. But did you launch the integration tests too ? I tried it quickly this morning before my day work and it failed too.

**Comment by Falydoor** [comment]
@pascalgrimaud Correct, working on it!

**Comment by Falydoor** [comment]
@pascalgrimaud So the test that is failing is `testGetUnknownAccount()` but it is not very logic to have it because there is no database call.

The only way to have an unknown account is to have a null `authentication` in the `SecurityContext` which technically can't happen since the method `getAccount` of `AccountResource` requires to be authenticated.

One way to keep the test would be to mock `SecurityUtils.getCurrentUserLogin()` to return null but it can't be done with Mockito since it's a static method.

So I'm kinda looking for ideas right now!

**Comment by pascalgrimaud** [comment]
> So the test that is failing is testGetUnknownAccount() but it is not very logic to have it because there is no database call.

as it's not logic, simply remove this test when there is no database, keep it for other option

File: generators/server/templates/src/test/java/package/web/rest/AccountResourceIT.java.ejs [hunk 1]
```diff
@@ -236,20 +236,6 @@ public class AccountResourceIT <% if (databaseType === 'cassandra') { %>extends
             .jsonPath("$.authorities").isEqualTo(AuthoritiesConstants.ADMIN);
         <%_ } _%>
     }
-
-    @Test
```
Comment (pascalgrimaud) [inline generators/server/templates/src/test/java/package/web/rest/AccountResourceIT.java.ejs]: @Falydoor : should it be excluded only for nodatabase and no oauth2 ?

Comment (Falydoor) [reply generators/server/templates/src/test/java/package/web/rest/AccountResourceIT.java.ejs]: The oauth2 part is in a different `if` and `nodatabase` is kinda linked with `skipUserManagement`.

I tested different scenarios and they were working but let me know if you have one that isn't working!

Comment (pascalgrimaud) [reply generators/server/templates/src/test/java/package/web/rest/AccountResourceIT.java.ejs]: Ok then, trusting you ! Thanks a lot @Falydoor

**Pull request merged**